samcli/commands/local/lib/local_lambda.py

```diff
@@ -246,13 +260,18 @@ def _make_env_vars(self, function):
             shell_env_values=shell_env,
             override_values=overrides,
             aws_creds=aws_creds,
-        )
+        )  # EnvironmentVariables is not yet annotated with type hints, disable mypy check for now. type: ignore
 
-    def _get_session_creds(self):
+    def _get_session_creds(self) -> Credentials:
         if self._boto3_session_creds is None:
             # to pass command line arguments for region & profile to setup boto3 default session
             LOG.debug("Loading AWS credentials from session with profile '%s'", self.aws_profile)
-            session = boto3.session.Session(profile_name=self.aws_profile, region_name=self.aws_region)
+            # The signature of boto3.session.Session defines the default values of profile_name and region_name
+            # so they should be Optional. But mypy follows its docstring which does not have "Optional."
+            # Here we trick mypy thinking they are both str rather than Optional[str].
+            session = boto3.session.Session(
+                profile_name=cast(str, self.aws_profile), region_name=cast(str, self.aws_region)
```

Comment:
Why we are using `cast(str, self.aws_profile)` instead of `str(self.aws_profile)` here?

Reply:
here we know `self.aws_profile` is `str`, `cast(str, self.aws_profile)` is effectively noop in runtime (return `self.aws_profile`) while `str(self.aws_profile)`, according to the doc

> Create a new string object from the given object.

Reply:
Are we using cast here, because `self.aws_profile` and `self.aws_region` are **Optional**[str] while mypy wants them to be str? if so, will this work
```
aws_profile = self.aws_profile # May be adding "or None" ??
aws_region = self.aws_region  # May be adding "or None" ??
session = boto3.session.Session(profile_name=aws_profile, region_name=aws_region)
```

Reply:
This won't work because it does not eliminate the "nullability," if we want to force `aws_profile` to be `str` instead of `Optional[str]`, there must be one of the below:

- `type: ignore`, like `aws_profile: str = self.aws_profile # type: ignore`
- `cast`, like `aws_profile = cast(str, self.aws_profile)`
- assertion, like `assert(self.aws_profile)` or `if not self.aws_profile: raise SomeException(.....)`